spinn_front_end_common/interface/interface_functions/application_runner.py

```diff
@@ -79,12 +58,25 @@ def __call__(
         if notification_interface is not None and wait_on_confirmation:
             notification_interface.wait_for_confirmation()
 
-        # set off the executables that are in sync state
-        if sync_signal is not None:
-            txrx.send_signal(app_id, sync_signal)
-            txrx.wait_for_cores_to_be_in_state(
-                executable_targets.all_core_subsets, app_id,
-                [CPUState.RUNNING, CPUState.PAUSED, CPUState.FINISHED])
+        # set off the executables that are in sync state \
+        # (sending to all is just as safe)
+        if (ExecutableType.USES_SIMULATION_INTERFACE in
+                executable_types.keys() or
```

Comment:
Don't use .keys to check the keys of a dict, as this expands it to a list...

Reply:
rubebred